pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -436,3 +445,119 @@ func TestAddNetworkPolicyWithMultipleRules(t *testing.T) {
 	assert.Equal(t, 2, controller.GetAddressGroupNum())
 	assert.Equal(t, 1, controller.GetAppliedToGroupNum())
 }
+
+func waitForReconcilerUpdated(t *testing.T, reconciler *mockReconciler) {
+	select {
+	case ruleID := <-reconciler.updated:
+		_, exists := reconciler.getLastRealized(ruleID)
+		if !exists {
+			t.Fatalf("Expected rule %s, got none", ruleID)
+		}
+	case <-time.After(time.Millisecond * 100):
+		t.Fatal("Expected one update, got none")
+	}
+}
+
+func waitForReconcilerDeleted(t *testing.T, reconciler *mockReconciler) {
+	select {
+	case ruleID := <-reconciler.deleted:
+		actualRule, exists := reconciler.getLastRealized(ruleID)
+		if exists {
+			t.Fatalf("Expected no rule, got %v", actualRule)
+		}
+	case <-time.After(time.Millisecond * 100):
+		t.Fatal("Expected one update, got none")
+	}
+}
```

Comment:
I think these could have been *local* functions / closures to leverage variable capture for `t` and `reconciler` (unless other tests start using them), but I'm nitpicking

Reply:
Thanks for the comment. It sounds good since I felt a bit weird when creating those functions with t and reconciler as arguments. 
I changed those 2 functions and checking networkpolicy metrics function into local functions since they are only used in this testing function. Please have a check if it looks good.